test/opt/set_spec_const_default_value_test.cpp

```diff
@@ -537,6 +541,519 @@ INSTANTIATE_TEST_CASE_P(
+        // 1. Empty with non-empty values to set.
+        {"", SpecIdToValueBitPatternMap{{1, {100}}, {2, {200}}}, ""},
+        // 2. Baisc bool type.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "%bool = OpTypeBool\n"
+            "%1 = OpSpecConstantTrue %bool\n"
+            "%2 = OpSpecConstantFalse %bool\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0x0}}, {101, {0x1}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "%bool = OpTypeBool\n"
+            "%1 = OpSpecConstantFalse %bool\n"
+            "%2 = OpSpecConstantTrue %bool\n",
+        },
+        // 3. 32-bit int type.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "OpDecorate %3 SpecId 102\n"
+            "%int = OpTypeInt 32 1\n"
+            "%1 = OpSpecConstant %int 10\n"
+            "%2 = OpSpecConstant %int 11\n"
+            "%3 = OpSpecConstant %int 11\n",
+            // default values
+            SpecIdToValueBitPatternMap{
+                {100, {2147483647}}, {101, {0xffffffff}}, {102, {0xffffffd6}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "OpDecorate %3 SpecId 102\n"
+            "%int = OpTypeInt 32 1\n"
+            "%1 = OpSpecConstant %int 2147483647\n"
+            "%2 = OpSpecConstant %int -1\n"
+            "%3 = OpSpecConstant %int -42\n",
+        },
+        // 4. 64-bit uint type.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "%ulong = OpTypeInt 64 0\n"
+            "%1 = OpSpecConstant %ulong 10\n"
+            "%2 = OpSpecConstant %ulong 11\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0xFFFFFFFE, 0xFFFFFFFF}},
+                                       {101, {0x100, 0x0}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "%ulong = OpTypeInt 64 0\n"
+            "%1 = OpSpecConstant %ulong 18446744073709551614\n"
+            "%2 = OpSpecConstant %ulong 256\n",
+        },
+        // 5. 32-bit float type.
+        {
+            // code
+            "OpDecorate %1 SpecId 101\n"
+            "OpDecorate %2 SpecId 102\n"
+            "%float = OpTypeFloat 32\n"
+            "%1 = OpSpecConstant %float 200\n"
+            "%2 = OpSpecConstant %float 201\n",
+            // default values
+            SpecIdToValueBitPatternMap{{101, {0xffffffff}},
+                                       {102, {0x40200000}}},
+            // expected
+            "OpDecorate %1 SpecId 101\n"
+            "OpDecorate %2 SpecId 102\n"
+            "%float = OpTypeFloat 32\n"
+            "%1 = OpSpecConstant %float -0x1.fffffep+128\n"
```

Comment:
Nice.  I'm so glad we were very careful with printing float numbers with the HexFloat class.  :-)